src/tree/Indented.js

```diff
@@ -0,0 +1,237 @@
+﻿"use strict";
+(function (root, factory) {
+    if (typeof define === "function" && define.amd) {
+        define(["d3", "../common/SVGZoomWidget", "../common/PropertyExt", "../api/ITree", "../common/Utility", "css!./Indented"], factory);
+    } else {
+        root.tree_Indented = factory(root.d3, root.common_SVGZoomWidget, root.common_PropertyExt, root.api_ITree, root.common_Utility);
+    }
+}(this, function (d3, SVGZoomWidget, PropertyExt, ITree, Utility) {
+    function Column(owner) {
+        PropertyExt.call(this);
+        this._owner = owner;
+    }
+    Column.prototype = Object.create(PropertyExt.prototype);
+    Column.prototype.constructor = Column;
+    Column.prototype._class += " tree_Dendrogram.Column";
+
+    Column.prototype.publish("column", null, "set", "Field", function () { return this._owner ? this._owner.columns() : []; }, { optional: true });
+
+    // ===
+    function Indented(target) {
+        SVGZoomWidget.call(this);
+        ITree.call(this);
+        Utility.SimpleSelectionMixin.call(this);
+
+        this._drawStartPos = "origin";
+
+        this._d3LayoutTree = d3.layout.tree();
+        this._d3Diagonal = d3.svg.diagonal()
+            .projection(function (d) {
+                return [d.y, d.x];
+            })
+        ;
+    }
+    Indented.prototype = Object.create(SVGZoomWidget.prototype);
+    Indented.prototype.constructor = Indented;
+    Indented.prototype._class += " tree_Indented";
+    Indented.prototype.implements(ITree.prototype);
+    Indented.prototype.mixin(Utility.SimpleSelectionMixin);
+    Indented.prototype.Column = Column;
+
+    Indented.prototype.publish("xmlColumn", null, "set", "Field", function () { return this.columns(); }, { optional: true });
+    Indented.prototype.publish("mappings", [], "propertyArray", "Source Columns", null, { autoExpand: Column, disable: function (w) { return w.xmlColumn_exists(); } });
+    Indented.prototype.publish("barHeight", 16, "number", "Bar height");
+
+    function xmlToJson(xml, id) {
+        id = id || "";
+        var retVal = {
+            id: id,
+            label: "",
+            attributes: {},
+            children: []
+        };
+
+        retVal.label = xml.nodeName;
+        if (xml.nodeType === 1) { // element
+            if (xml.attributes.length > 0) {
+                for (var j = 0; j < xml.attributes.length; j++) {
+                    var attribute = xml.attributes.item(j);
+                    retVal.attributes[attribute.nodeName] = attribute.nodeValue;
+                }
+            }
+        } else if (xml.nodeType === 3) { // text
+            retVal.label = xml.nodeValue;
+        }
+
+        if (xml.hasChildNodes()) {
+            for (var i = 0; i < xml.childNodes.length; i++) {
+                var item = xml.childNodes.item(i);
+                var child = xmlToJson(item, id + "[" + retVal.children.length + "]");
```

Comment:
yes - but not ideal, the ID is primarily to help with the animation (so the DOM nodes are re-used correctly and consistently)